src/librbd/ImageWatcher.cc

```diff
@@ -844,7 +844,6 @@ void ImageWatcher::handle_payload(const FlattenPayload &payload,
 
 void ImageWatcher::handle_payload(const ResizePayload &payload,
 				  bufferlist *out) {
-  RWLock::RLocker l(m_image_ctx.owner_lock);
```

Comment:
> In this case, w/o the owner lock being held, it's possible for the exclusive lock to be released before the op has a chance to add itself to the pending list.

Exclusive lock is released -> add op to the pending list -> start a new async resize request -> the state machine detects the op is cancelled or the owner lock is lost and returns with -ERESTART -> the callback function `RemoteContext::finish` schedules async complete -> `notify_async_complete` notifies remote and removes the op from pending list

In this scenario, we are still able to remove the op from the pending list and keep things consistent. Do you mean that we shouldn't put an op in the pending list after the exclusive lock is released? Or am I wrong somewhere?